server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRecipientRewriteTableMailet.java

```diff
@@ -19,22 +19,28 @@
 

```

Comment:
It's not about sharing code I think. I agree with you with common function will be on common service.
But in this case, I think it's not really common functions. There are 2 mailets with the same process (in this case: UsersRepositoryAliasingForwarding and RecipientRewriteTable). Its service must changeReturnPath --> buildNewRecipient --> processMail --> .... and each subclass has its processMail so I think the abstract class is acceptable in case of there are same process. WDYT?